README
MultiEdit is a plugin originally created for version 1.8 of Qgis. 
This plugin let the user choose an attribute value from a vector 
layer and selecting the matching features,allows the user change 
all the corresponding to a new value.
It is particurarly useful if you have many features to modify 
assigning the same value or if you have the necessity to write 
the same value to a new field in the table of attributes of 
your vector layer wherever there's a matching feature to your
criterion: this plugin lets you write the new value in any 
existing field in the attribute table,for the selected 
features or in any other newly created fields directly 
from the plugin gui.
This branch is intended to be the future developing ground
and at the moment is compatible with 3.x version of Qgis.

Giuseppe De Marco


CHANGELOG
1.1
MultiEdit v. 1.1 2018 release
Major updates to match python3 and PyQT5 for QGis new 3.x realeases. Use of QgsComboBox custom widgets for layers and fields parsing, code cleanup and simplified functions. 
1.0
MultiEdit v. 1.0 2015 release
Major updates for Qgis 2.6 (or higher) release and API changes. Added function to select all features and change values for all by the new Select all pushbutton. 

0.5
Major changes in function to meet new API
Major changes in plugin loading to match new menu/toolbar specs of Qgis >=1.9

0.4 
Fixed strange behaviour in functions after plugin unloading and reloading
*Added About/help first version

0.3
Fixed select features function
Update comboboxes after saving
Update related comboboxes on combobox item change

0.2.1
Fixed bug in create new field feature

0.2
Fixed minor bugs and issues of version 0.1: 
Qlinedit instead Qplaintext,cleaning of older code snippets
*added function to add new fields to write values to!


0.1
First release

....
no earlier entries


about.py
# -*- coding: utf-8 -*-

# Form implementation generated from reading ui file 'about.ui'
#
# Created by: PyQt5 UI code generator 5.5.1
#
# WARNING! All changes made in this file will be lost!

from PyQt5 import QtCore, QtGui, QtWidgets

class Ui_About(object):
    def setupUi(self, About):
        About.setObjectName("About")
        About.resize(810, 559)
        self.label_3 = QtWidgets.QLabel(About)
        self.label_3.setGeometry(QtCore.QRect(280, 0, 191, 41))
        font = QtGui.QFont()
        font.setPointSize(15)
        font.setBold(True)
        font.setWeight(75)
        self.label_3.setFont(font)
        self.label_3.setAlignment(QtCore.Qt.AlignCenter)
        self.label_3.setObjectName("label_3")
        self.label = QtWidgets.QLabel(About)
        self.label.setGeometry(QtCore.QRect(220, 40, 311, 41))
        font = QtGui.QFont()
        font.setPointSize(9)
        self.label.setFont(font)
        self.label.setAlignment(QtCore.Qt.AlignCenter)
        self.label.setObjectName("label")
        self.textBrowser = QtWidgets.QTextBrowser(About)
        self.textBrowser.setGeometry(QtCore.QRect(20, 100, 761, 151))
        self.textBrowser.setObjectName("textBrowser")
        self.label_2 = QtWidgets.QLabel(About)
        self.label_2.setGeometry(QtCore.QRect(240, 10, 51, 51))
        self.label_2.setText("")
        self.label_2.setPixmap(QtGui.QPixmap(":/plugins/MultiEdit/pienocampo.png"))
        self.label_2.setScaledContents(True)
        self.label_2.setObjectName("label_2")
        self.webView = QtWebKitWidgets.QWebView(About)
        self.webView.setGeometry(QtCore.QRect(10, 250, 771, 271))
        self.webView.setUrl(QtCore.QUrl("qrc:/plugins/MultiEdit/singlehtml/singlehtml/index.html"))
        self.webView.setZoomFactor(0.699999988079071)
        self.webView.setObjectName("webView")

        self.retranslateUi(About)
        QtCore.QMetaObject.connectSlotsByName(About)

    def retranslateUi(self, About):
        _translate = QtCore.QCoreApplication.translate
        About.setWindowTitle(_translate("About", "Dialog"))
        self.label_3.setText(_translate("About", "MultiEdit v 1.1"))
        self.label.setText(_translate("About", "(c) 2018 Giuseppe De Marco\n"
" d2gis\n"
" www.d2gis.com"))
        self.textBrowser.setHtml(_translate("About", "<!DOCTYPE HTML PUBLIC \"-//W3C//DTD HTML 4.0//EN\" \"http://www.w3.org/TR/REC-html40/strict.dtd\">\n"
"<html><head><meta name=\"qrichtext\" content=\"1\" /><style type=\"text/css\">\n"
"p, li { white-space: pre-wrap; }\n"
"</style></head><body style=\" font-family:\'Ubuntu\'; font-size:11pt; font-weight:400; font-style:normal;\">\n"
"<p style=\" margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;\">MultiEdit v. 1.1 April 2018 release</p>\n"
"<p style=\" margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;\">Major updates for Qgis 3.x release and API changes. Added function to select all features and change values for all by the new Select all pushbutton. </p>\n"
"<p style=\" margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;\">This plugins\' new releases are currently ongoing.</p>\n"
"<p style=\"-qt-paragraph-type:empty; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;\"><br /></p></body></html>"))

from PyQt5 import QtWebKitWidgets
import resources_rc


__init__.py
"""
/***************************************************************************
 MultiEdit
                                 A QGIS plugin
 MultiEdit chooses an attribute value from a vector layer and selecting the matching features, changes all the corresponding to a new value: useful if you have many features to modify.This plugin lets you write the new value in any existing field in the attribute table, for the selected features or newly created from plugin gui...
                             -------------------
        begin                : 2012-03-09
        copyright            : (C) 2018 by Giuseppe De Marco
        email                : info@pienocampo.it
 ***************************************************************************/

/***************************************************************************
 *                                                                         *
 *   This program is free software; you can redistribute it and/or modify  *
 *   it under the terms of the GNU General Public License as published by  *
 *   the Free Software Foundation; either version 2 of the License, or     *
 *   (at your option) any later version.                                   *
 *                                                                         *
 ***************************************************************************/
 This script initializes the plugin, making it known to QGIS.
"""
from __future__ import absolute_import
def name():
    return "MultiEdit"
def description():
    return "MultiEdit chooses an attribute value from a vector layer and selecting the matching features, changes all the corresponding to a new value: useful if you have many features to modify.This plugin lets you write the new value in any existing field in the attribute table, for the selected features or newly created from plugin gui..."
def version():
    return "Version 1.1"
def icon():
    return "multiedit.png"
def qgisMinimumVersion():
    return "3.0"
def classFactory(iface):
    # load MultiEdit class from file MultiEdit
    from .multiedit import MultiEdit
    return MultiEdit(iface)


multieditdialog.py
"""
/***************************************************************************
 MultiEditDialog
                                 A QGIS plugin
 MultiEdit chooses an attribute value from a vector layer and selecting the matching features, changes all the corresponding to a new value: useful if you have many features to modify.This plugin lets you write the new value in any existing field in the attribute table, for the selected features or newly created from plugin gui...
                             -------------------
        begin                : 2012-03-09
        copyright            : (C) 2012 by Giuseppe De Marco
        email                : info@pienocampo.it
 ***************************************************************************/

/***************************************************************************
 *                                                                         *
 *   This program is free software; you can redistribute it and/or modify  *
 *   it under the terms of the GNU General Public License as published by  *
 *   the Free Software Foundation; either version 2 of the License, or     *
 *   (at your option) any later version.                                   *
 *                                                                         *
 ***************************************************************************/
"""

from qgis.PyQt import QtCore, QtGui, QtWidgets
from .ui_multiedit import Ui_MultiEdit
# create the dialog for zoom to point
class MultiEditDialog(QtWidgets.QDialog):
    def __init__(self):
        QtWidgets.QDialog.__init__(self)
        # Set up the user interface from Designer.
        self.ui = Ui_MultiEdit()
        self.ui.setupUi(self)


ui_multiedit.py
# -*- coding: utf-8 -*-

# Form implementation generated from reading ui file 'ui_multiedit.ui'
#
# Created by: PyQt5 UI code generator 5.5.1
#
# WARNING! All changes made in this file will be lost!

from PyQt5 import QtCore, QtGui, QtWidgets

class Ui_MultiEdit(object):
    def setupUi(self, MultiEdit):
        MultiEdit.setObjectName("MultiEdit")
        MultiEdit.resize(801, 517)
        font = QtGui.QFont()
        font.setFamily("Verdana")
        MultiEdit.setFont(font)
        self.label = QtWidgets.QLabel(MultiEdit)
        self.label.setGeometry(QtCore.QRect(20, 60, 67, 17))
        self.label.setObjectName("label")
        self.label_2 = QtWidgets.QLabel(MultiEdit)
        self.label_2.setGeometry(QtCore.QRect(20, 119, 67, 17))
        self.label_2.setObjectName("label_2")
        self.oldattribute = QtWidgets.QComboBox(MultiEdit)
        self.oldattribute.setGeometry(QtCore.QRect(10, 200, 261, 27))
        self.oldattribute.setInsertPolicy(QtWidgets.QComboBox.InsertAlphabetically)
        self.oldattribute.setObjectName("oldattribute")
        self.label_3 = QtWidgets.QLabel(MultiEdit)
        self.label_3.setGeometry(QtCore.QRect(20, 180, 141, 17))
        self.label_3.setObjectName("label_3")
        self.label_4 = QtWidgets.QLabel(MultiEdit)
        self.label_4.setGeometry(QtCore.QRect(300, 180, 161, 17))
        self.label_4.setObjectName("label_4")
        self.txtFeedBack = QtWidgets.QTextBrowser(MultiEdit)
        self.txtFeedBack.setGeometry(QtCore.QRect(10, 360, 781, 131))
        self.txtFeedBack.setObjectName("txtFeedBack")
        self.line = QtWidgets.QFrame(MultiEdit)
        self.line.setGeometry(QtCore.QRect(10, 50, 841, 20))
        self.line.setFrameShadow(QtWidgets.QFrame.Sunken)
        self.line.setFrameShape(QtWidgets.QFrame.HLine)
        self.line.setObjectName("line")
        self.label_5 = QtWidgets.QLabel(MultiEdit)
        self.label_5.setGeometry(QtCore.QRect(0, 0, 351, 41))
        self.label_5.setAlignment(QtCore.Qt.AlignCenter)
        self.label_5.setObjectName("label_5")
        self.select = QtWidgets.QPushButton(MultiEdit)
        self.select.setGeometry(QtCore.QRect(20, 240, 111, 41))
        self.select.setObjectName("select")
        self.unselect = QtWidgets.QPushButton(MultiEdit)
        self.unselect.setGeometry(QtCore.QRect(140, 240, 121, 41))
        self.unselect.setObjectName("unselect")
        self.change_another = QtWidgets.QPushButton(MultiEdit)
        self.change_another.setGeometry(QtCore.QRect(440, 250, 171, 41))
        self.change_another.setObjectName("change_another")
        self.label_6 = QtWidgets.QLabel(MultiEdit)
        self.label_6.setGeometry(QtCore.QRect(293, 59, 261, 17))
        self.label_6.setObjectName("label_6")
        self.label_7 = QtWidgets.QLabel(MultiEdit)
        self.label_7.setGeometry(QtCore.QRect(10, 343, 81, 20))
        self.label_7.setObjectName("label_7")
        self.Exit = QtWidgets.QPushButton(MultiEdit)
        self.Exit.setGeometry(QtCore.QRect(620, 310, 171, 41))
        self.Exit.setObjectName("Exit")
        self.save = QtWidgets.QPushButton(MultiEdit)
        self.save.setGeometry(QtCore.QRect(617, 250, 171, 41))
        self.save.setObjectName("save")
        self.show_t = QtWidgets.QPushButton(MultiEdit)
        self.show_t.setGeometry(QtCore.QRect(440, 310, 171, 41))
        self.show_t.setObjectName("show_t")
        self.fieldtype = QtWidgets.QComboBox(MultiEdit)
        self.fieldtype.setGeometry(QtCore.QRect(577, 138, 71, 27))
        self.fieldtype.setObjectName("fieldtype")
        self.fieldtype.addItem("")
        self.fieldtype.addItem("")
        self.fieldtype.addItem("")
        self.create_new_field = QtWidgets.QPushButton(MultiEdit)
        self.create_new_field.setGeometry(QtCore.QRect(650, 138, 141, 27))
        self.create_new_field.setObjectName("create_new_field")
        self.line_6 = QtWidgets.QFrame(MultiEdit)
        self.line_6.setGeometry(QtCore.QRect(283, 290, 511, 20))
        self.line_6.setFrameShadow(QtWidgets.QFrame.Sunken)
        self.line_6.setFrameShape(QtWidgets.QFrame.HLine)
        self.line_6.setObjectName("line_6")
        self.label_8 = QtWidgets.QLabel(MultiEdit)
        self.label_8.setGeometry(QtCore.QRect(584, 60, 201, 17))
        self.label_8.setObjectName("label_8")
        self.label_10 = QtWidgets.QLabel(MultiEdit)
        self.label_10.setGeometry(QtCore.QRect(580, 120, 161, 17))
        self.label_10.setObjectName("label_10")
        self.newvalue = QtWidgets.QLineEdit(MultiEdit)
        self.newvalue.setGeometry(QtCore.QRect(296, 201, 491, 27))
        self.newvalue.setObjectName("newvalue")
        self.newfield = QtWidgets.QLineEdit(MultiEdit)
        self.newfield.setGeometry(QtCore.QRect(574, 80, 220, 27))
        self.newfield.setObjectName("newfield")
        self.label_12 = QtWidgets.QLabel(MultiEdit)
        self.label_12.setGeometry(QtCore.QRect(350, 10, 51, 41))
        self.label_12.setText("")
        self.label_12.setPixmap(QtGui.QPixmap(":/plugins/MultiEdit/pienocampo.png"))
        self.label_12.setScaledContents(True)
        self.label_12.setObjectName("label_12")
        self.line_3 = QtWidgets.QFrame(MultiEdit)
        self.line_3.setGeometry(QtCore.QRect(270, 60, 20, 271))
        self.line_3.setFrameShape(QtWidgets.QFrame.VLine)
        self.line_3.setFrameShadow(QtWidgets.QFrame.Sunken)
        self.line_3.setObjectName("line_3")
        self.line_2 = QtWidgets.QFrame(MultiEdit)
        self.line_2.setGeometry(QtCore.QRect(560, 60, 20, 111))
        self.line_2.setFrameShape(QtWidgets.QFrame.VLine)
        self.line_2.setFrameShadow(QtWidgets.QFrame.Sunken)
        self.line_2.setObjectName("line_2")
        self.line_4 = QtWidgets.QFrame(MultiEdit)
        self.line_4.setGeometry(QtCore.QRect(570, 166, 221, 16))
        self.line_4.setFrameShape(QtWidgets.QFrame.HLine)
        self.line_4.setFrameShadow(QtWidgets.QFrame.Sunken)
        self.line_4.setObjectName("line_4")
        self.about = QtWidgets.QPushButton(MultiEdit)
        self.about.setGeometry(QtCore.QRect(610, 10, 171, 41))
        self.about.setObjectName("about")
        self.all = QtWidgets.QPushButton(MultiEdit)
        self.all.setGeometry(QtCore.QRect(20, 290, 111, 21))
        self.all.setObjectName("all")
        self.line_7 = QtWidgets.QFrame(MultiEdit)
        self.line_7.setGeometry(QtCore.QRect(10, 320, 271, 20))
        self.line_7.setFrameShadow(QtWidgets.QFrame.Sunken)
        self.line_7.setFrameShape(QtWidgets.QFrame.HLine)
        self.line_7.setObjectName("line_7")
        self.label_9 = QtWidgets.QLabel(MultiEdit)
        self.label_9.setGeometry(QtCore.QRect(400, 20, 131, 21))
        self.label_9.setText("")
        self.label_9.setPixmap(QtGui.QPixmap(":/plugins/MultiEdit/d2gis.com.png"))
        self.label_9.setScaledContents(True)
        self.label_9.setObjectName("label_9")
        self.chosenlayer = QgsMapLayerComboBox(MultiEdit)
        self.chosenlayer.setGeometry(QtCore.QRect(70, 60, 211, 31))
        self.chosenlayer.setInsertPolicy(QtWidgets.QComboBox.InsertAlphabetically)
        self.chosenlayer.setObjectName("chosenlayer")
        self.Column = QgsFieldComboBox(MultiEdit)
        self.Column.setGeometry(QtCore.QRect(69, 110, 211, 31))
        self.Column.setInsertPolicy(QtWidgets.QComboBox.InsertAlphabetically)
        self.Column.setObjectName("Column")
        self.new_field = QgsFieldComboBox(MultiEdit)
        self.new_field.setGeometry(QtCore.QRect(290, 80, 241, 27))
        self.new_field.setObjectName("new_field")
        self.label.raise_()
        self.label_2.raise_()
        self.oldattribute.raise_()
        self.label_3.raise_()
        self.label_4.raise_()
        self.txtFeedBack.raise_()
        self.line.raise_()
        self.select.raise_()
        self.unselect.raise_()
        self.change_another.raise_()
        self.label_6.raise_()
        self.label_7.raise_()
        self.Exit.raise_()
        self.save.raise_()
        self.show_t.raise_()
        self.fieldtype.raise_()
        self.create_new_field.raise_()
        self.line_6.raise_()
        self.label_8.raise_()
        self.label_10.raise_()
        self.newvalue.raise_()
        self.newfield.raise_()
        self.label_12.raise_()
        self.line_3.raise_()
        self.line_2.raise_()
        self.line_4.raise_()
        self.about.raise_()
        self.all.raise_()
        self.line_7.raise_()
        self.label_9.raise_()
        self.label_5.raise_()
        self.chosenlayer.raise_()
        self.Column.raise_()
        self.new_field.raise_()

        self.retranslateUi(MultiEdit)
        self.Exit.clicked['bool'].connect(MultiEdit.reject)
        QtCore.QMetaObject.connectSlotsByName(MultiEdit)

    def retranslateUi(self, MultiEdit):
        _translate = QtCore.QCoreApplication.translate
        MultiEdit.setWindowTitle(_translate("MultiEdit", "MultiEdit"))
        self.label.setText(_translate("MultiEdit", "Layer"))
        self.label_2.setText(_translate("MultiEdit", "Field"))
        self.label_3.setText(_translate("MultiEdit", "Old Attribute value"))
        self.label_4.setText(_translate("MultiEdit", "New Attribute value"))
        self.label_5.setText(_translate("MultiEdit", "MultiEdit 1.1  (C) 2018 by Giuseppe De Marco   \n"
"www.d2gis.com"))
        self.select.setText(_translate("MultiEdit", "Select\n"
"Features"))
        self.unselect.setText(_translate("MultiEdit", "Clear Selected\n"
"Features"))
        self.change_another.setText(_translate("MultiEdit", "Write value \n"
" to chosen field"))
        self.label_6.setText(_translate("MultiEdit", "Field to write the new value to:"))
        self.label_7.setText(_translate("MultiEdit", "Messages"))
        self.Exit.setText(_translate("MultiEdit", "Exit"))
        self.save.setText(_translate("MultiEdit", "Save Changes"))
        self.show_t.setText(_translate("MultiEdit", "Show Attribute Table \n"
"of selected Layer"))
        self.fieldtype.setItemText(0, _translate("MultiEdit", "String"))
        self.fieldtype.setItemText(1, _translate("MultiEdit", "Int"))
        self.fieldtype.setItemText(2, _translate("MultiEdit", "Double"))
        self.create_new_field.setText(_translate("MultiEdit", "Create new Field"))
        self.label_8.setText(_translate("MultiEdit", "New Field Name"))
        self.label_10.setText(_translate("MultiEdit", "Field type"))
        self.about.setText(_translate("MultiEdit", "About/Help"))
        self.all.setText(_translate("MultiEdit", "Select All"))

from qgis.gui import QgsFieldComboBox
from qgis.gui import QgsMapLayerComboBox
import resources_rc
